projects/dxilconv/include/ShaderBinary/ShaderBinary.h

```diff
@@ -2582,8 +2582,8 @@ class CShaderCodeParser
         CShaderToken* pEndTok = m_pShaderEndToken;
         CShaderToken* pRet;
 
-        m_pCurrentToken = (CShaderToken*)pBuffer;
-        m_pShaderEndToken = (CShaderToken*)pBufferEnd;
+        m_pCurrentToken = const_cast<CShaderToken*>(pBuffer);
+        m_pShaderEndToken = const_cast<CShaderToken*>(pBufferEnd);
```

Comment:
When I encountered const losing cast warnings before, I took the opportunity to investigate how they are used and see if something can be constified instead of just casting the const off in a way that silences the error. I may have missed something, but my examination of `m_pCurrentToken` and `m_pShaderEndToken` suggest that they can be made const. That would be a better way to resolve this though it may require some additional changes.

Reply:
Fixed by add const to m_pCurrentToken and m_pShaderEndToken.